kratos/python_scripts/multistage_orchestrators/sequential_multistage_orchestrator.py

```diff
@@ -0,0 +1,175 @@
+            else:
+                KratosMultiphysics.Logger.PrintInfo("'execution_list' is not provided. Stages will be executed according to their declaration order.")
+                self._execution_list = list(self.GetProject().GetSettings()["stages"].keys())
+
+        return self._execution_list
+
+    def __GetStagesToCheckpointList(self):
+        '''Creates and returns the stages checkpoint list.
+
+        This method creates the stages checkpoint list, either from a user-defined checkpoint list or,
+        if a bool is provided, it creates a list with all the stages in the execution list.
+        '''
+
+        # Check if the checkpoint stages list has been already created
+        if not hasattr(self, "_stages_to_checkpoint_list"):
+            if self.GetProject().GetSettings()["orchestrator"]["settings"]["stage_checkpoints"].IsBool():
+                if self.GetProject().GetSettings()["orchestrator"]["settings"]["stage_checkpoints"].GetBool():
+                    # All stages are to be checkpointed
+                    self._stages_to_checkpoint_list = self.__GetExecutionList()
+                else:
+                    # Create an empty list to avoid errors
+                    self._stages_to_checkpoint_list = []
+
+            elif self.GetProject().GetSettings()["orchestrator"]["settings"]["stage_checkpoints"].IsStringArray():
+                # Check that the stages asked to be checkpointed are in the execution list
+                stages_asked_to_checkpoint = self.GetProject().GetSettings()["orchestrator"]["settings"]["stage_checkpoints"].GetStringArray()
+                aux_diff = set(stages_asked_to_checkpoint) - set(self.__GetExecutionList())
+                if len(aux_diff) == 0:
+                    self._stages_to_checkpoint_list = stages_asked_to_checkpoint
+                else:
+                    err_msg = f"User asked to checkpoint stages that are not present in 'execution_list'. Please remove these {list(aux_diff)}."
+                    raise ValueError(err_msg)
+            else:
+                err_msg = "'stage_checkpoints' value must be bool or a list with the stages to be checkpointed."
+                raise TypeError(err_msg)
+        
+        return self._stages_to_checkpoint_list
+    
+    def __GetStagesCheckpointFolder(self):
+        '''Gets the folder to store the checkpoint files from the settings.'''
+
+        if self.GetProject().GetSettings()["orchestrator"]["settings"].Has("stage_checkpoints_folder"):
+            return self.GetProject().GetSettings()["orchestrator"]["settings"]["stage_checkpoints_folder"].GetString()
+        else:
+            warn_msg = "'stage_checkpoints_folder' is not provided. Creating a default 'checkpoints' one."
+            KratosMultiphysics.Logger.PrintWarning(warn_msg)
+            return 'checkpoints'
+        
+    def __PrepareOutputSettings(self, stage_name : str):
```

Comment:
Shouldn't be PrepareCheckPointSettings?

Reply:
mmm... I don't think so. The checkpoint settings are a tiny portion of all the settings we are outputting.

Reply:
I don't get the design then. What are the outputs?

Reply:
The name of the output settings file (e.g. `ProjectParametersOutput.json`) . Also note that the function edits current settings accordingly.